### PR TITLE
[IPCT1-281] update image uploading to use presigned url

### PR DESCRIPTION
**Pull request by obernardovieira**
### Related
This relates to https://github.com/impactMarket/api/pull/63

### Video/Pictures
result video [jump to 1:08 to start] https://drive.google.com/file/d/1IfvdxF4JoE7I7QwEUBHpAYDL8n5zMXYD/view?usp=sharing
note: the video was recorded on the low-end SLIDE device. Which crashed at first.

### Tests
No new tests. Previous tests were broken on master.

### CODE REVIEW

File: src/views/createCommunity/index.tsx [hunk 1]
```diff
@@ -603,94 +592,44 @@ function CreateCommunityScreen() {
...
+                    if (community !== undefined) {
+                        batch(() => {
+                            dispatch(setCommunityMetadata(community));
+                            dispatch(setUserIsCommunityManager(true));
+                        });
                     }
-                    setSending(false);
-                    setSendingSuccess(true);
                 }
+                setSending(false);
```
Comment (Jonathansoufer) [inline src/views/createCommunity/index.tsx:629]: Duplicated State changes ???

File: src/services/api/routes/community.ts [hunk 2]
```diff
@@ -122,14 +126,92 @@ class ApiRouteCommunity {
...
+            httpMethod: 'PUT',
+            mimeType: mimetype,
+            uploadType: 0, //FileSystemUploadType.BINARY_CONTENT
+            headers: {
+                'Content-Type': 'image/' + mimetype,
+            },
+        });
+        if (ru.status >= 400) {
+            throw new Error(ru.body.toString());
+        }
```
Comment (Jonathansoufer) [inline src/services/api/routes/community.ts:151]: Is there any error treatment/handler when throwing this error?

**Review by Jonathansoufer — CHANGES_REQUESTED** [review]

File: src/services/api/routes/community.ts (hunk 2 above)
Comment (obernardovieira) [reply src/services/api/routes/community.ts:151]: Yes.
The whole process is surrounded by try/catch. https://github.com/impactMarket/mobile-app/pull/112/files/e2c8bbb8a91ed6606dd2d58dd46ff5166c082726#diff-37cf8af5e15d4449d5297985e2c1e148f148b90af3ad2b2e0c20cb326424f742L694

File: src/views/createCommunity/index.tsx (hunk 1 above)
Comment (obernardovieira) [reply src/views/createCommunity/index.tsx:629]: ops. Thanks

**Pull request merged**